chimedb/dataflag/client.py

```diff
@@ -144,11 +196,393 @@ def convert(self, value, param, ctx):
 # =========================
 
 
+@db.atomic(read_write=True)
```

Comment:
@ketiltrout I guess this is mostly a question for you. Given that some of the operations below are interactive and may be expected to take a while, what happens if they take too long? Does MySQL timeout the transaction and then it fails when it tries to commit anything?

Reply:
You mean because of the `click.confirm()`s?  That's a good question.  I'm not sure.  It may just block forever.

Reply:
Indeed, that was what I was thinking of.

I think it's probably not an issue provided it doesn't place a lock on the whole DB during the atomic section. Presumably it's just using a transaction such that it doesn't need to.